### PR TITLE
Add role voter

### PR DESCRIPTION
**Pull request by itkg-nanne**
[OO-FEATURE] add voter on role usage
[OO-BUGFIX] fix error on role delete

https://github.com/open-orchestra/open-orchestra-user-bundle/pull/94
https://github.com/open-orchestra/open-orchestra-model-interface/pull/171
https://github.com/open-orchestra/open-orchestra-workflow-function-bundle/pull/92
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1530

### CODE REVIEW

File: Backoffice/Security/Authorization/Voter/DeleteRoleVoter.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
+<?php
```
Comment (alavieille) [inline Backoffice/Security/Authorization/Voter/DeleteRoleVoter.php]: can you add test for this class

File: Backoffice/UsageFinder/RoleUsageFinder.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+class RoleUsageFinder
+{
+    protected $RoleableElementRepositories = array();
+
+    /**
+     * @param RoleableElementRepositoryInterface $repository
+     */
+    public function addRepository(RoleableElementRepositoryInterface $repository)
+    {
+        $this->RoleableElementRepositories[] = $repository;
```
Comment (alavieille) [inline Backoffice/UsageFinder/RoleUsageFinder.php]: RoleableElementRepositories => roleableElementRepositories

File: Backoffice/UsageFinder/RoleUsageFinder.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace OpenOrchestra\Backoffice\UsageFinder;
```
Comment (alavieille) [inline Backoffice/UsageFinder/RoleUsageFinder.php]: can you add test for this class

**Comment by alavieille** [comment]
the OO-TAG is not very explicit

**Comment by itkg-nanne** [comment]
update done @alavieille

File: Backoffice/Security/Authorization/Voter/DeleteRoleVoter.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    /**
+     * Returns the vote for the given parameters.
+     *
+     * This method must return one of the following constants:
+     * ACCESS_GRANTED, ACCESS_DENIED, or ACCESS_ABSTAIN.
+     *
+     * @param TokenInterface              $token      A TokenInterface instance
+     * @param RoleInterface|object|null $object     The object to secure
```
Comment (alavieille) [inline Backoffice/Security/Authorization/Voter/DeleteRoleVoter.php]: align parameter

**Comment by alavieille** [comment]
else align parameter :+1: and when test run

**Pull request merged**